pkg/network/node/ovs/parse.go

```diff
@@ -397,3 +414,131 @@ func UnparseExternalIDs(externalIDs map[string]string) string {
 	}
 	return "{" + strings.Join(ids, ",") + "}"
 }
+
+func ParseGroup(group string) (*OVSGroup, error) {
+	// According to the man page "Such flow descriptions comprise a series  field=value
+	// assignments,  separated  by  commas or white space.  (Embedding spaces into a
+	// group description normally requires quoting to prevent the  shell  from breaking
+	// the description into multiple arguments.)."
+	parsed := &OVSGroup{}
+
+	origGroup := group
+	for group != "" {
+		field := ""
+		value := ""
+		end := strings.IndexAny(group, ", ")
+		if end == -1 {
+			end = len(group)
+		}
+		eq := strings.Index(group, "=")
+		if eq == -1 || eq > end {
+			field = group[:end]
+		} else {
+			field = group[:eq]
+			if field == "bucket" {
+				//find the index of the next bucket to determine the end
+				end = strings.Index(group[eq+1:], "bucket")
```

Comment:
It's confusing to use `end` for this, since everywhere else in the loop, `end` is a length relative to `group`, but here it's a length relative to `group[eq+1:]`. Should use a different temporary variable instead (`nextBucket`?)

Reply:
done